include/mutex.h

```diff
@@ -0,0 +1,577 @@
+/****************************************************************************
+ * include/mutex.h
```

Comment:
mutex.h is too general, it's easy to conflict with the 3rd party library or source code, if we put iit into include folder.